exercises/Daniel_Murphy/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,126 @@
+#include <iostream>
+#include <fstream>
+#include <map>
+#include <string>
+#include <vector>
+#include <algorithm>
+#include <functional>
+#include <iomanip>
+#include <sstream>
+
+std::vector<std::string> AnalyseLine(std::string line)
+{
+  // Remove punctuation that accompanies a space
+  std::vector<char> punctuation = {'?','.',',', '\'', '"', '(', ')', '!', ':', ';'};
+  for (char mark : punctuation)
+  {
+    line.erase(std::remove(line.begin(), line.end(),mark), line.end());
+  }
+
+  // Hyphens don't accompany spaces so replace with a space and split words
+  std::replace( line.begin(), line.end(), '-', ' ');
+
+  // force all letters to case lower
+  std::transform(line.begin(), line.end(), line.begin(), [](unsigned char c){ return std::tolower(c);});
+
+  std::vector<std::string> words;
+
+  std::stringstream lineOfText(line);
+  std::string word;
+  while (lineOfText >> word) {
+    if (word.size() > 4)
+      words.push_back(word);
+  }
+
+  return words;
+}
+
+std::map<std::string, int> UpdateMap(const std::vector<std::string> &words, std::map<std::string,int>& Counter)
+{
+  for (auto& word : words)
+  {
+    // if word already a member, increment its count by 1
+    Counter[word]++ ;
+  }
+  return Counter;
```

Comment:
Thanks for changing this. Now that you're passing Counter by reference you don't need a return type in the function signature nor a return statement here. L92 will also need adjusting if you change this